workflow/controller/workflowpod.go

```diff
@@ -695,83 +704,98 @@ func addSchedulingConstraints(pod *apiv1.Pod, wfSpec *wfv1.WorkflowSpec, tmpl *w
 	}
 }
 
-// addVolumeReferences adds any volumeMounts that a container/sidecar is referencing, to the pod.spec.volumes
-// These are either specified in the workflow.spec.volumes or the workflow.spec.volumeClaimTemplate section
-func addVolumeReferences(pod *apiv1.Pod, vols []apiv1.Volume, tmpl *wfv1.Template, pvcs []apiv1.Volume) error {
-	switch tmpl.GetType() {
-	case wfv1.TemplateTypeContainer, wfv1.TemplateTypeScript:
-	default:
-		return nil
-	}
-
+func (woc *wfOperationCtx) addVolumeRef(volMounts []apiv1.VolumeMount, pod *apiv1.Pod, tmpl *wfv1.Template) error {
```

Comment:
can I ask you re-order this func and move it below `addVolumeReferences`? it'll make it easier to identify what is new code and what is old code